### PR TITLE
Introduced ability to mix and match YAML and XML configs for Jet and …

### PR DESCRIPTION
**Pull request by eminn**
…IMDG

Fixes https://github.com/hazelcast/hazelcast-jet/issues/1472

### CODE REVIEW

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/config/ConfigProvider.java [hunk 1]
```diff
@@ -102,4 +105,35 @@ public static ClientConfig locateAndGetClientConfig() {
...
+
+    @Nonnull
+    public static Config locateAndGetMemberConfig(@Nullable Properties properties) {
+        Config config;
+        XmlJetMemberConfigLocator xmlConfigLocator = new XmlJetMemberConfigLocator();
+        YamlJetMemberConfigLocator yamlConfigLocator = new YamlJetMemberConfigLocator();
+
+        if (yamlConfigLocator.locateFromSystemProperty()) {
+            // 1. Try loading config if provided in system property and it is an YAML file
+            config = new YamlConfigBuilder(yamlConfigLocator.getIn()).setProperties(properties).build();
```
Comment (gurbuzali) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/config/ConfigProvider.java]: does yaml have priority over xml ?

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/config/YamlJetMemberConfigLocator.java [hunk 2]
```diff
@@ -30,7 +30,7 @@
     private static final String HAZELCAST_ENTERPRISE_MEMBER_DEFAULT_YAML = "hazelcast-jet-enterprise-member-default.yaml";
 
     public YamlJetMemberConfigLocator() {
-        super(true);
+        super(false);
```
Comment (gurbuzali) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/config/YamlJetMemberConfigLocator.java]: I guess this means `dont fail if the suffix is not expected`, why ?

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/config/ConfigProvider.java (hunk 1 above)
Comment (eminn) [reply hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/config/ConfigProvider.java]: This is related to https://github.com/hazelcast/hazelcast-jet/issues/1380, if it doesn't have yaml extensions we are treating it as XML one. So we are checking if the file has yaml extensions first.

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/config/YamlJetMemberConfigLocator.java (hunk 2 above)
Comment (eminn) [reply hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/config/YamlJetMemberConfigLocator.java]: It shouldn't fail to be able to fallback to other methods like loading from classpath etc.

**Pull request merged**